CMakeLists.txt

```diff
@@ -480,14 +480,36 @@ function(add_picolibc directory variant target_triple flags qemu_params variant_
         tls
         ungetc
     )
-    #remove tests that fail at least at one platform
-    list(REMOVE_ITEM picolibc_tests abort constructor-skip long_double malloc malloc_stress math_errhandling test-except timegm tls)
     # remove tests failing on a given platform
     if(variant STREQUAL "armv8m.main_hard_fp")
-        list(REMOVE_ITEM picolibc_tests complex-funcs fenv math-funcs printf_scanf rand test-efcvt test-strtod)
+        list(REMOVE_ITEM picolibc_tests abort constructor-skip complex-funcs fenv long_double math_errhandling math-funcs malloc_stress printf_scanf rand test-efcvt test-except test-strtod timegm tls)
     endif()
     if(variant STREQUAL "armv8.1m.main_hard_nofp_mve")
-        list(REMOVE_ITEM picolibc_tests complex-funcs fenv ffs math-funcs printf_scanf printf-tests rand regex test-efcvt test-strtod time-tests)
+        list(REMOVE_ITEM picolibc_tests abort constructor-skip complex-funcs fenv ffs long_double math_errhandling math-funcs malloc_stress printf_scanf printf-tests rand regex test-efcvt test-except test-strtod timegm time-tests tls)
+    endif()
+    if(variant STREQUAL "armv8.1m.main_hard_fp")
+        list(REMOVE_ITEM picolibc_tests abort constructor-skip long_double math_errhandling malloc_stress test-except timegm tls)
+    endif()
+    if(variant STREQUAL "armv8.1m.main_soft_nofp_nomve")
+        list(REMOVE_ITEM picolibc_tests abort constructor-skip long_double math_errhandling malloc_stress test-except timegm tls)
+    endif()
+    if(variant STREQUAL "armv8m.main_soft_nofp")
+        list(REMOVE_ITEM picolibc_tests abort constructor-skip long_double math_errhandling malloc_stress test-except timegm tls)
+    endif()
+    if(variant STREQUAL "armv7m_soft_nofp")
+        list(REMOVE_ITEM picolibc_tests abort constructor-skip test-except tls)
+    endif()
+    if(variant STREQUAL "armv7em_soft_nofp")
+        list(REMOVE_ITEM picolibc_tests abort constructor-skip test-except tls)
+    endif()
+    if(variant STREQUAL "armv7em_hard_fpv4_sp_d16")
+        list(REMOVE_ITEM picolibc_tests abort constructor-skip math_errhandling test-except tls)
+    endif()
+    if(variant STREQUAL "armv7em_hard_fpv5_d16")
+        list(REMOVE_ITEM picolibc_tests abort constructor-skip math_errhandling test-except tls)
+    endif()
+    if(variant STREQUAL "armv6m_soft_nofp")
+        list(REMOVE_ITEM picolibc_tests abort constructor-skip test-except tls)
     endif()
```

Comment:
Nit: please sort these in a way that is consistent with the list passed to add_library_variants()

Reply:
It's sorted in same way as list of variants is sorted.